### DISCUSSION TITLE
Parcel fails to resolve 'solid-js/jsx-runtime' for TSX files

### DISCUSSION THREAD

**Discussion by pmwmedia** [solidjs/solid#884] [Q&A]
I'm migrating a React Webpack project into a SolidJS Parcel project. However, the combination of SolidJS and Parcel seems to have problems with nested Typescript projects, which the combination of React and Webpack doesn't have.

My `package.json` in the root projects has the most dependencies and the `package.json` in the child foo project has the build command. When I try to exceute the parcel build, I get the error: `@parcel/core: Failed to resolve 'solid-js/jsx-runtime' from './foo/src/index.tsx'`. However, if I rename, `index.tsx` into `index.jsx`, Parcel can build the same file without any error. Nevertheless, I want to use Typescript and not JavaScript.

/.babelrc:
```json
{
    "presets": [ "solid" ]
}
```

/package.json:
```json
{
    "private": true,
    "dependencies": {
        "solid-js": "1.3.12"
    },
    "devDependencies": {
        "@babel/core": "7.17.7",
        "babel-preset-solid": "1.3.12",
        "parcel": "2.3.2",
        "typescript": "4.6.2"
    }
}
```

/tsconfig.json:
```json
{
    "compilerOptions": {
        "isolatedModules": true,
        "jsx": "preserve",
        "jsxImportSource": "solid-js",
        "lib": [
            "dom",
            "es2021"
        ],
        "module": "commonjs",
        "noEmit": true,
        "noImplicitAny": true,
        "noImplicitReturns": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "sourceMap": true,
        "strict": true,
        "target": "es5"
    }
}
```

/foo/src/index.tsx:
```jsx
export function HelloWorld() {
    return <div>Hello World!</div>;
}
```

/foo/package.json:
```json
{
    "private": true,
    "source": "src/index.tsx",
    "scripts": {
        "parcel:build": "npx parcel build"
    }
}
```

I have uploaded all files in an example repository https://github.com/pmwmedia/solidjs-example including a GitHub workflow that reproduces the error.

How can I fix the error `@parcel/core: Failed to resolve 'solid-js/jsx-runtime` in my nested Typescript project?

**Comment by CoolOppo** [comment]
I'm having the same issue, but I don't have multiple projects going like you do. Will update if I figure it out.

**Comment by lxsmnsyc** [comment]
This is a frequent issue on Parcel. The issue is that there are versions where the babel config is loaded correctly. I'm not sure what causes this issue though.

**Comment by TiagoCavalcante** [comment]
@pmwmedia did you fixed it?

**Reply by pmwmedia** [reply]
I found no way to fix it with parcel. Therefore, I migrated to webpack, what works without issues.

**Reply by TiagoCavalcante** [reply]
Oh, I found how to fix it now.

It error happens because "jsx-runtime" is TS only, and the bundle should be JS only. To fix it you only need to add a .babelrc with the following:
```json
{ "presets": ["solid"] }
```

And run `npm i babel-preset-solid`.

**Reply by pmwmedia** [reply]
In my case, it didn't help. In my original post above, you can see that I had such .babelrc file.

**Reply by danieltroger** [reply]
For me, removing `/** @jsxImportSource solid-js */` from the top of my file fixed the issue

**Reply by danieltroger** [reply]
Err that actually made it not use solids jsx runtime
